rasa_nlu/featurizers/regex_featurizer.py

```diff
@@ -72,18 +72,19 @@ def features_for_patterns(self, message):
         relating the name of the regex to whether it was matched."""
 
         matches = []
-        for i, exp in enumerate(self.known_patterns):
-            match = re.search(exp["pattern"], message.text)
-            matches.append(match)
+        for exp in self.known_patterns:
+            match = re.finditer(exp["pattern"], message.text)
+            match_item = [mat for mat in match]
```

Comment:
I doubt you need this line, you can iterate over the iterator, you do not need a list (and if you do, please use `list(match)` instead of the generator)

Reply:
You can only iterate over the iterator once that's why i had to save it in a list. I'll change it to ``list(match)``